### PR TITLE
Add Recipe for Flax

### PR DESCRIPTION
**Pull request by PhilipVinc**
Adds [google/flax](https://github.com/google/flax) to Conda-forge.

As discussed with @jheek in [Flax#1778](https://github.com/google/flax/issues/1178)

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by PhilipVinc** [comment]
According to Conda-forge rules, I ask the contributors named by @jheek to acknowledge that they will be listed as maintainers and code-owners of the recipe.

> GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
@jheek
@marcvanzee 
@avital
@levskaya 
@andsteing

**Comment by jheek** [comment]
> GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
I confirm

**Comment by avital** [comment]
> GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.

I confirm.

**Comment by PhilipVinc** [comment]
@marcvanzee, @levskaya,  @andsteing Could you please write a comment aknowledging that you'll be listed as maintainer and code owner of the conda recipe for Flax as requested by jheek?

Sorry for spamming

File: recipes/flax/meta.yaml
Comment (dopplershift) [inline recipes/flax/meta.yaml:24]: Unless you need Qt for use with matplotlib, best to depend on `matplotlib-base`.

File: recipes/flax/meta.yaml
Comment (dopplershift) [inline recipes/flax/meta.yaml:13]: It looks like this package is pure python? If so, you can remove the skip, add `noarch: python` here and then change the dependencies on Python to `python >=3.6`.

What's the need to skip on Windows? If it's just a dependency is unavailable, we can just ignore that.

File: recipes/flax/meta.yaml
Comment (PhilipVinc) [reply recipes/flax/meta.yaml]: fixed.

File: recipes/flax/meta.yaml
Comment (PhilipVinc) [reply recipes/flax/meta.yaml]: it was an unavailable dependency. I addressed the comment by making the recipe noarch.

Comment (PhilipVinc) [reply recipes/flax/meta.yaml]: Any idea why the windows build is failing with `conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform win-64` ? 
Jaxlib is not available for windows so it should skip it.

**Comment by levskaya** [comment]
> GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.

I confirm

**Comment by avital** [comment]
@marcvanzee is on parental leave for the next 10 days, so maybe we should
take him off the maintainer list right now and add him later to unblock
this?

**Comment by PhilipVinc** [comment]
@avital done. We can indeed add him later. Only @andsteing is still missing.

@dopplershift I have made the recipe no arch as you requested, but the windows builder is still failing with an unsatisfiable Dependency Error, which shouldn't happen because a transitive dependency, jaxlib, is not available on windows. Do you have any suggestions?

**Comment by PhilipVinc** [comment]
@andsteing we are waiting for you to comment in order to merge this one

File: recipes/flax/meta.yaml
Comment (dopplershift) [reply recipes/flax/meta.yaml]: That's the error it spits out when a dependency is not available. It doesn't skip, it just fails CI--but we can ignore to merge.

**Comment by PhilipVinc** [comment]
@dopplershift ready for final review/merge

**Comment by andsteing** [comment]
I confirm

**Review by dopplershift — APPROVED** [review]

**Pull request merged**
